pysteps/datasets.py

```diff
@@ -0,0 +1,353 @@
+            )
+    else:
+        os.makedirs(dir_path)
+
+    # NOTE:
+    # The http response from github can either contain Content-Length (size of the file)
+    # or use chunked Transfer-Encoding.
+    # If Transfer-Encoding is chunked, then the Content-Length is not available since
+    # the content is dynamically generated and we can't know the length a pr_iori easily.
+    pbar = ShowProgress()
+    request.urlretrieve(
+        "https://github.com/pySTEPS/pysteps-data/archive/master.zip",
+        tmp_file.name,
+        pbar,
+    )
+    pbar.end()
+
+    with ZipFile(tmp_file.name, "r") as zip_obj:
+        tmp_dir = TemporaryDirectory()
+
+        # Extract all the contents of zip file in the temp directory
+        common_path = os.path.commonprefix(zip_obj.namelist())
+
+        zip_obj.extractall(tmp_dir.name)
+
+        copy_tree(os.path.join(tmp_dir.name, common_path), dir_path)
+
+
+def create_default_pystepsrc(pysteps_data_dir, config_dir=None, file_name="pystepsrc"):
+    """
+    Create a default configuration file pointing to the pysteps data directory.
+
+    If the configuration file already exists, it backup the existing file by appending
+    to the filename the extensions '.1', '.2', up to '.5.'.
+    A maximum of 5 files are kept. .2, up to app.log.5.
+
+    File rotation is implemented for the backup files.
+    For example, if the default configuration filename is 'pystepsrc' and the files
+    pystepsrc, pystepsrc.1, pystepsrc.2, etc. exist, they are renamed to pystepsrc.1,
+    pystepsrc.2, pystepsrc.2, etc. respectively. Finally, after the existing files are
+    backed up, the new configuration file is written.
+    
+    Parameters
+    ----------
+
+    pysteps_data_dir : str
+        Path to the directory with the pysteps data.
+
+    config_dir : str
+        Destination directory for the configuration file.
+        Default values: $HOME/.pysteps (unix and Mac OS X)
+        or $USERPROFILE/pysteps (windows).
+        The directory is created if it does not exists.
+
+    file_name : str
+        Configuration file name. `pystepsrc` by default.
+
+    Returns
+    -------
+
+    dest_path : str
+        Configuration file path.
+    """
+
+    pysteps_lib_root = os.path.dirname(
+        _decode_filesystem_path(pysteps.__file__)
+    )
+
+    # Load the library built-in configuration file
+    with open(os.path.join(pysteps_lib_root, "pystepsrc"), "r") as f:
+        rcparams_json = json.loads(jsmin(f.read()))
+
+    for key, value in rcparams_json["data_sources"].items():
+        value["root_path"] = os.path.abspath(
+            os.path.join(pysteps_data_dir, value["root_path"])
+        )
+
+    if config_dir is None:
+        home_dir = os.path.expanduser("~")
+        if os.name == "nt":
+            subdir = "pysteps"
+        else:
+            subdir = ".pysteps"
+        config_dir = os.path.join(home_dir, subdir, file_name)
```

Comment:
This will create a `$HOME/.pysteps/pystepsrc/` directory where to save the new config file, which will have path `$HOME/.pysteps/pystepsrc/pystepsrc`.
By doing so, the new pystepsrc won't be found the next time pysteps is imported, since it will look for a `$HOME/.pysteps/pystepsrc` file instead. 
I suggest changing line 259 into `config_dir = os.path.join(home_dir, subdir)`.

Reply:
Solved in 689dc01.